### PR TITLE
Documentation enhancement

### PR DESCRIPTION
**Pull request by ChristianBeilschmidt**

### CODE REVIEW

File: projects/wave-core/src/lib/colors/color-breakpoint.model.ts [hunk 1]
```diff
@@ -19,56 +19,83 @@ export class ColorBreakpoint implements ColorBreakpointDict {
...
 
+    /**
+     * Compares the ColorBreakpoint with another one.
+     */
     equals(other: ColorBreakpoint): boolean {
         return other && this.rgba.equals(other.rgba) && this.value === other.value;
     }
 
+    /**
+     * Transforms the ColorBreakpoint into a representation understood by Mappings colorizer.
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/colors/color-breakpoint.model.ts:89]:
```suggestion
     * Transforms the ColorBreakpoint into a representation understood by Mapping's colorizer.
```

File: projects/wave-core/src/lib/colors/color-breakpoint.model.ts [hunk 2]
```diff
@@ -19,56 +19,83 @@ export class ColorBreakpoint implements ColorBreakpointDict {
...
 
+    /**
+     * Returns true if the value is a number.
+     */
     valueIsNumber(): boolean {
         return typeof this.value === 'number';
     }
 
+    /**
+     * Compares the ColorBreakpoint with another one.
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/colors/color-breakpoint.model.ts:82]:
```suggestion
     * Tests a ColorBreakpoint for equality with another one.
```

File: projects/wave-core/src/lib/colors/colorizer-data.model.ts [hunk 3]
```diff
@@ -3,31 +3,47 @@ import {Color} from './color';
 
 /**
  * DEPRECATED
+ * TODO: remove
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/colors/colorizer-data.model.ts:6]:
```suggestion
```
why does deprecation need another TODO?

File: projects/wave-core/src/lib/colors/colorizer-data.model.ts [hunk 4]
```diff
@@ -81,38 +100,62 @@ export class ColorizerData implements IColorizerData {
...
 
+    /**
+     * Removes all breakpoints.
+     */
     clear() {
         this.breakpoints = [];
     }
 
+    /**
+     * Adds a breakpoint at the end of the list.
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/colors/colorizer-data.model.ts:136]:
```suggestion
     * Adds a breakpoint to the end of the list.
```

File: projects/wave-core/src/lib/colors/colorizer-data.model.ts [hunk 5]
```diff
@@ -186,10 +232,16 @@ export class ColorizerData implements IColorizerData {
         return (!this.breakpoints || this.isEmpty()) ? undefined : this.breakpoints[this.breakpoints.length - 1];
     }
 
+    /**
+     * Checks id the list of breakpoints is empty.
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/colors/colorizer-data.model.ts:236]:
```suggestion
     * Checks if the list of breakpoints is empty.
```

File: projects/wave-core/src/lib/colors/colorizer-data.model.ts [hunk 6]
```diff
@@ -186,10 +232,16 @@ export class ColorizerData implements IColorizerData {
...
+     */
     isEmpty(): boolean {
         return !this.breakpoints || this.breakpoints.length === 0;
     }
 
+    /**
+     * Update the tyoe of the ColorizerData.
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/colors/colorizer-data.model.ts:243]:
```suggestion
     * Update the type of the ColorizerData.
```

File: projects/wave-core/src/lib/colors/colorizer-editor/colorizer-editor.component.ts [hunk 7]
```diff
@@ -12,6 +12,9 @@ import {ColorizerData, ColorizerType} from '../colorizer-data.model';
 import {NG_VALUE_ACCESSOR, ControlValueAccessor} from '@angular/forms';
 import {ColorBreakpoint} from '../color-breakpoint.model';
 
+/**
+ * The ColorizerEditorComponent is the main ui for editing ColorizerData / ColorBreakpoints
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/colors/colorizer-editor/colorizer-editor.component.ts:16]:
```suggestion
 * The ColorizerEditorComponent is the main dialog for editing ColorizerData / ColorBreakpoints
```

File: projects/wave-core/src/lib/colors/colorizer-editor/colorizer-editor.component.ts [hunk 8]
```diff
@@ -83,6 +122,9 @@ export class ColorizerEditorComponent implements ControlValueAccessor, OnChanges
         }
     }
 
+    /**
+     * Add a new breakpoint a tposition i. Clones the next breakpoint if possible.
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/colors/colorizer-editor/colorizer-editor.component.ts:126]:
```suggestion
     * Add a new breakpoint at position i. Clones the next breakpoint if possible.
```

File: projects/wave-core/src/lib/colors/colormap-colorizer/colormap-colorizer.component.ts [hunk 9]
```diff
@@ -21,6 +21,9 @@ import {FormBuilder, FormGroup, Validators} from '@angular/forms';
 import {valueRelation, WaveValidators} from '../../util/form.validators';
 import {Subscription} from 'rxjs';
 
+/**
+ * The ColormapColorizerComponent is the main ui to generate ColorizerData from colormaps.
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/colors/colormap-colorizer/colormap-colorizer.component.ts:25]:
```suggestion
 * The ColormapColorizerComponent is a dialog to generate ColorizerData from colormaps.
```

File: projects/wave-core/src/lib/colors/colormap-colorizer/colormap-colorizer.component.ts [hunk 10]
```diff
@@ -29,19 +32,56 @@ import {Subscription} from 'rxjs';
...
     readonly colormapNames = COLORMAP_NAMES;
     readonly boundedColormapStepScales = COLORMAP_STEP_SCALES_WITH_BOUNDS;
+
+    /**
+     * The form control used in the template.
+     */
     form: FormGroup;
+
+    /**
+     * The local (wip) ColorizerData.
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/colors/colormap-colorizer/colormap-colorizer.component.ts:80]:
```suggestion
     * The local (work-in-progress) ColorizerData.
```

File: projects/wave-core/src/lib/colors/colormap-colorizer/colormap-colorizer.component.ts [hunk 11]
```diff
@@ -138,6 +185,9 @@ export class ColormapColorizerComponent implements OnInit, OnDestroy, OnChanges
         this.colorizerData = colorizerData;
     }
 
+    /**
+     * Apply a new colortable to the colorizer data.
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/colors/colormap-colorizer/colormap-colorizer.component.ts:189]:
```suggestion
     * Apply a new color table to the colorizer data.
```

File: projects/wave-core/src/lib/colors/colormap-colorizer/colormap-name-to-colorizer-data.pipe.ts [hunk 12]
```diff
@@ -2,7 +2,9 @@ import {Pipe, PipeTransform} from '@angular/core';
 import {IColorizerData} from '../colorizer-data.model';
 import {Colormap, ColormapNames, ColormapStepScale} from '../colormaps/colormap.model';
 
-
+/**
+ * A pipe to transform a colormap into a colorizer data.
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/colors/colormap-colorizer/colormap-name-to-colorizer-data.pipe.ts:6]:
```suggestion
 * A pipe to transform a color map into a colorizer data.
```

File: projects/wave-core/src/lib/colors/colormaps/colormap.model.ts [hunk 13]
```diff
@@ -43,28 +43,54 @@ import {
 } from './scientific-colormaps/scientific-colormaps';
 import {colormap_rainbow_data, GENERIC_COLORMAP_NAMES, GenericColormapName} from './generic-colormaps';
 
+/**
+ * Type for ColormapData / a wrapper for RGB values.
+ */
 export type ColormapData = Array<[number, number, number]>;
+
+/**
+ * All allowed colormap names.
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/colors/colormaps/colormap.model.ts:52]:
```suggestion
 * All allowed `Colormap` names.
```

File: projects/wave-core/src/lib/colors/colormaps/colormap.model.ts [hunk 14]
```diff
@@ -43,28 +43,54 @@ import {
...
+ */
 export type ColormapNames = MplColormapName | MorelandColormapName | ScientificColormapName | GenericColormapName;
+
+/**
+ * A list of all colormap names.
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/colors/colormaps/colormap.model.ts:57]:
```suggestion
 * A list of all `Colormap` names.
```

File: projects/wave-core/src/lib/colors/colormaps/colormap.model.ts [hunk 15]
```diff
@@ -43,28 +43,54 @@ import {
...
+ */
 export const COLORMAP_NAMES: Array<ColormapNames> = [...MPL_COLORMAP_NAMES, ...MORELAND_COLORMAP_NAMES, ...SCIENTIFIC_COLORMAP_NAMES,
     ...GENERIC_COLORMAP_NAMES];
 
+/**
+ * The colormap step scaling methods.
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/colors/colormaps/colormap.model.ts:63]:
```suggestion
 * The `Colormap` step scaling methods.
```

File: projects/wave-core/src/lib/colors/colormaps/colormap.model.ts [hunk 16]
```diff
@@ -43,28 +43,54 @@ import {
...
+ */
 export const COLORMAP_STEP_SCALES_WITH_BOUNDS: Array<BoundedColormapStepScale> = [
     {stepScaleName: 'linear'},
     {stepScaleName: 'log', requiresValueAbove: 0},
     {stepScaleName: 'square root', requiresValueBelow: 5000},
     {stepScaleName: 'square', requiresValueBelow: 5000}
 ];
 
+/**
+ * Abstract class for common colormap functions.
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/colors/colormaps/colormap.model.ts:87]:
```suggestion
 * Abstract class for common `Colormap` functions.
```

File: projects/wave-core/src/lib/colors/colormaps/colormap.model.ts [hunk 17]
```diff
@@ -43,28 +43,54 @@ import {
...
+ */
 export abstract class Colormap {
 
+    /**
+     * Resolves the colormap data for a colormap name.
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/colors/colormaps/colormap.model.ts:92]:
```suggestion
     * Resolves the `Colormap` data for a `Colormap` name.
```

File: projects/wave-core/src/lib/colors/colormaps/generic-colormaps.ts [hunk 18]
```diff
@@ -1,6 +1,16 @@
+/**
+ * The colormap name.
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/colors/colormaps/generic-colormaps.ts:2]:
```suggestion
 * The color map name.
```

File: projects/wave-core/src/lib/colors/colormaps/generic-colormaps.ts [hunk 19]
```diff
@@ -1,6 +1,16 @@
...
+ */
 export type GenericColormapName = 'RAINBOW';
+
+/**
+ * The list of available colormap names.
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/colors/colormaps/generic-colormaps.ts:7]:
```suggestion
 * The list of available color map names.
```

File: projects/wave-core/src/lib/colors/colormaps/generic-colormaps.ts [hunk 20]
```diff
@@ -1,6 +1,16 @@
...
+ */
 export const GENERIC_COLORMAP_NAMES: Array<GenericColormapName> = ['RAINBOW'];
 
+/**
+ * The rainbow colormap.
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/colors/colormaps/generic-colormaps.ts:12]:
```suggestion
 * The rainbow color map.
```

File: projects/wave-core/src/lib/colors/colormaps/moreland-colormaps.ts [hunk 21]
```diff
@@ -3,12 +3,21 @@ import {ColormapNames} from './colormap.model';
 /**
  * This color maps are from CoolWarmFloat33.csv of "Diverging Color Maps for Scientific Visualization" by Kenneth Moreland
  * on <http://www.kennethmoreland.com/color-maps/>
- **/
+ */
 
+/**
+ * The colormap name.
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/colors/colormaps/moreland-colormaps.ts:9]:
```suggestion
 * The color map name.
```

File: projects/wave-core/src/lib/colors/colormaps/moreland-colormaps.ts [hunk 22]
```diff
@@ -3,12 +3,21 @@ import {ColormapNames} from './colormap.model';
...
+ */
 export type MorelandColormapName = 'COOLWARM';
-export const MORELAND_COLORMAP_NAMES: Array<MorelandColormapName> = ['COOLWARM'];
 
+/**
+ * The list of available colormap names.
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/colors/colormaps/moreland-colormaps.ts:14]:
```suggestion
 * The list of available color map names.
```

File: projects/wave-core/src/lib/colors/colormaps/moreland-colormaps.ts [hunk 23]
```diff
@@ -3,12 +3,21 @@ import {ColormapNames} from './colormap.model';
...
+ */
+export const MORELAND_COLORMAP_NAMES: Array<MorelandColormapName> = ['COOLWARM'];
 
+/**
+ * The coolwarm colormap.
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/colors/colormaps/moreland-colormaps.ts:19]:
```suggestion
 * The coolwarm color map.
```

File: projects/wave-core/src/lib/colors/colormaps/mpl-colormaps.ts [hunk 24]
```diff
@@ -5,7 +5,15 @@
  * They are licensed under the CC0 license.
  */
 
+
+ /**
+  * The colormap names.
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/colors/colormaps/mpl-colormaps.ts:10]:
```suggestion
  * The color map names.
```

File: projects/wave-core/src/lib/colors/colormaps/mpl-colormaps.ts [hunk 25]
```diff
@@ -5,7 +5,15 @@
...
+  */
 export type MplColormapName = 'MAGMA' | 'INFERNO' | 'PLASMA' | 'VIRIDIS';
+
+/**
+ * The list of available colormap names.
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/colors/colormaps/mpl-colormaps.ts:15]:
```suggestion
 * The list of available color map names.
```

File: projects/wave-core/src/lib/colors/colormaps/scientific-colormaps/scientific-colormaps.ts [hunk 26]
```diff
@@ -6,15 +6,23 @@
  *      Geosci. Model Dev., 11, 2541-2562, doi:10.5194/gmd-11-2541-2018
  *
  * See 'The Scientific Colour Maps' MIT license and copyright in scientific-colormaps.license
- **/
-export type ScientificColormapName = 'ARCON' | 'BAMAKO' | 'BATLOW' | 'BERLIN' | 'BILBAO' | 'BROC' | 'BROCO'
+ */
+
+/**
+ * The colormap names.
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/colors/colormaps/scientific-colormaps/scientific-colormaps.ts:12]:
```suggestion
 * The color map names.
```

File: projects/wave-core/src/lib/colors/colormaps/scientific-colormaps/scientific-colormaps.ts [hunk 27]
```diff
@@ -6,15 +6,23 @@
...
+ */
+ export type ScientificColormapName = 'ARCON' | 'BAMAKO' | 'BATLOW' | 'BERLIN' | 'BILBAO' | 'BROC' | 'BROCO'
     | 'BUDA' | 'CORC' | 'CORCO' | 'DAVOS' | 'DEVON' | 'GRAYC' | 'HAWAII' | 'IMOLA' | 'LAJOLLA' | 'LAPAZ'
     | 'LISBON' | 'NUUK' | 'OLERON' | 'OSLO' | 'ROMA' | 'ROMAO' | 'TOFINO' | 'TOKYO' | 'TURKU' | 'VIK' | 'VIKO';
-export const SCIENTIFIC_COLORMAP_NAMES: Array<ScientificColormapName> = ['ARCON', 'BAMAKO', 'BATLOW', 'BERLIN', 'BILBAO', 'BROC',
+
+/**
+ * The list of available colormap names.
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/colors/colormaps/scientific-colormaps/scientific-colormaps.ts:19]:
```suggestion
 * The list of available color map names.
```

File: projects/wave-core/src/lib/operators/operator-type-parameter-options.model.ts [hunk 28]
```diff
@@ -234,9 +293,21 @@ export class StringParameterArray extends AbstractParameterContainer<string> {
     }
 }
 
+/**
+ * An interface for numberic range parameters.
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/operators/operator-type-parameter-options.model.ts:297]:
```suggestion
 * An interface for numeric range parameters.
```

File: projects/wave-core/src/lib/project/project.service.ts [hunk 29]
```diff
@@ -790,7 +842,7 @@ export class ProjectService {
                     this.notificationService.error(`${plot.name}: ${reason.status} ${reason.statusText}`);
                     loadingState$.next(LoadingState.ERROR);
                 }
-            ),)
+            ), )
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/project/project.service.ts:845]: Im not convinced of these changes

File: projects/wave-core/src/lib/util/pipes/mapping-colorizer-to-gradient.pipe.ts [hunk 30]
```diff
@@ -2,14 +2,23 @@ import {Pipe, PipeTransform} from '@angular/core';
...
 import {Color} from '../../colors/color';
 
+/**
+ * Pipe to transform colorizer data into a css string.
+ */
 @Pipe({name: 'waveWappingColorizerToGradient'})
 export class MappingColorizerToGradientPipe implements PipeTransform {
 
+    /**
+     * Transform r,g,b,a values into a css string.
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/util/pipes/mapping-colorizer-to-gradient.pipe.ts:12]:
```suggestion
     * Transform red, green, blue, and alpha values into a css string.
```

File: projects/wave-core/src/lib/project/project.service.ts (hunk 29 above)
Comment (jdroenner) [reply projects/wave-core/src/lib/project/project.service.ts:845]: tslint suggested this

**Pull request merged**